pallets/loans/src/loan.rs

```diff
@@ -164,7 +164,7 @@ impl<T: Config> Pallet<T> {
 
         exchange_rate
             .checked_mul_int(collateral_factor * collateral)
-            .and_then(|r| r.checked_mul(currency_price))
+            .and_then(|r| r.checked_mul(currency_price.into_inner()))
```

Comment:
The function `collateral_asset_value` can return `Result<FixedU128, Error>`, which can reduce the use of `into_inner`.

Reply:
Fixed